### PR TITLE
documentation: include `migration` guide

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

The pull-request introduces a `migration guide` to the framework to help downstream extensions and applications migrate to newer versions of the framework.

#### How to test

- verify documentation content.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
I'll leave the pull-request as a draft to see if anyone has any feedback over the content, or format of the initial migration guide of the framework.

**Review by tsmaeder — APPROVED** [review]
Good start! As we discussed in the community call, the migration guide is supposed to be a living document initiated by the feature author and improved as adopters migrate to new releases.

**Comment by vince-fugnitto** [comment]
@paul-marechal should we add some entry about what you discovered, mainly how you can opt to not use the pre-built binaries for  `keytar` and instead build it for your operating system if you run into issues?

- https://github.com/eclipse-theia/theia/pull/9707/files#diff-be1156289fcb3cf2ac512dbea9fafba086b6d1494cfa8b90a850734a04f6e5f4R82-R94

**Comment by vince-fugnitto** [comment]
@marcdumais-work I think the initial version of the document is good, it should be updated when adopters notice additional migration steps that might be required after releases :+1:

File: doc/Migration.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+[Release](https://github.com/eclipse-theia/theia/releases/tag/v1.15.0)
+
+#### Keytar:
+
+- [`keytar`](https://github.com/atom/node-keytar) was added as a dependency for the secrets API. and may require `libsecret` in your particular distribution to be functional:
+  - Debian/Ubuntu: `sudo apt-get install libsecret-1-dev`
+  - Red Hat-based: `sudo yum install libsecret-devel`
+  - Arch Linux: `sudo pacman -S libsecret`
+- It is possible that a `yarn resolution` is necessary for `keytar` to work on older distributions (the fix was added in `1.16.0` by downgrading the dependency version):
```
Comment (azatsarynnyy) [inline doc/Migration.md:26]: For rebuilding under Alpine, it requires `apk add libsecret-dev`.
We faced it in Che-Theia. So, I believe it's worth mentioning as well.

**Comment by vince-fugnitto** [comment]
@azatsarynnyy I updated the migration and the prerequisites to include an `alpine` entry for `keytar`.

**Comment by azatsarynnyy** [comment]
> @azatsarynnyy I updated the migration and the prerequisites to include an `alpine` entry for `keytar`.

Thank you @vince-fugnitto!

**Pull request merged**